src/lib/utils.py

```diff
@@ -15,7 +15,7 @@
 from functools import partial, reduce
 from typing import Any, Callable, List, Dict, Tuple, Optional
 from numpy import unique
-from pandas import DataFrame, Series, concat, merge
+from pandas import DataFrame, Series, concat, merge, isnull
```

Comment:
Please use `lib.cast.isna` instead (it's speed-optimized)

Reply:
Done.